apps/frontend/src/app/PageTeam/CharacterDisplay/CustomMultiTarget/CustomMultiTargetCard.tsx

```diff
@@ -0,0 +1,326 @@
+            <Box display="flex" gap={2}>
+              <Button
+                onClick={onDup}
+                color="info"
+                sx={{ flexGrow: 1 }}
+                startIcon={<ContentCopyIcon />}
+              >
+                {t('multiTarget.duplicate')}
+              </Button>
+              <Button
+                color="info"
+                onClick={copyToClipboard}
+                startIcon={<ContentPasteIcon />}
+                sx={{ flexGrow: 1 }}
+              >
+                Export
+              </Button>
+              <Button color="error" onClick={onDelete}>
+                <DeleteForeverIcon />
+              </Button>
+            </Box>
+          </CardContent>
+          <Divider />
+          <CardContent
+            sx={{ display: 'flex', flexDirection: 'column', gap: 1 }}
+          >
+            {customTargetDisplays}
+            <AddCustomTargetBtn setTarget={addTarget} />
+          </CardContent>
+        </CardThemed>
+      </ModalWrapper>
+    </>
+  )
+}
+function NameDesc({
+  name: nameProp,
+  desc: descProp,
+  setTarget,
+}: {
+  name: string
+  desc?: string
+  setTarget: Dispatch<SetStateAction<CustomMultiTarget>>
+}) {
+  const [name, setName] = useState(nameProp)
+  const nameDeferred = useDeferredValue(name)
+  const [desc, setDesc] = useState(descProp)
+  const descDeferred = useDeferredValue(desc)
+
+  useEffect(() => {
+    setName(nameProp)
+  }, [nameProp])
+
+  useEffect(() => {
+    setDesc(descProp)
+  }, [descProp])
+
+  useEffect(() => {
+    setTarget((target) => ({
+      ...target,
+      name: nameDeferred,
+    }))
+    // Don't need to trigger when teamCharId is changed, only when the name is changed.
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [setTarget, nameDeferred])
+
+  useEffect(() => {
+    setTarget((target) => ({
+      ...target,
+      description: descDeferred,
+    }))
+    // Don't need to trigger when teamCharId is changed, only when the name is changed.
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [setTarget, descDeferred])
+
+  return (
+    <>
+      <TextField
+        fullWidth
+        label="Custom Multi-target Name"
+        value={name}
+        onChange={(e) => setName(e.target.value)}
+      />
+      <TextField
+        fullWidth
+        label="Custom Multi-target Description"
+        value={desc}
+        onChange={(e) => setDesc(e.target.value)}
+        multiline
+        minRows={2}
+      />
+    </>
+  )
+}
+
+function AddCustomTargetBtn({
+  setTarget,
+}: {
+  setTarget: (t: string[], m?: number) => void
+}) {
+  const { t } = useTranslation('page_character')
+  const [show, onShow, onClose] = useBoolState(false)
+  const setTargetHandler = useCallback(
+    (target: string[], multi?: number) => {
+      onClose()
+      setTarget(target, multi)
+    },
+    [onClose, setTarget]
+  )
+
+  return (
+    <>
+      <Button
```

Comment:
Should we update any new buttons to be outlined to save some migration effort later?

Reply:
Probably will get addressed later, need to see how the final layout look like.